internal/internal_event_handlers.go

```diff
@@ -1311,7 +1356,20 @@ func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 				err = weh.dataConverter.FromPayloads(sideEffectDataPayload, &sideEffectDataID, &sideEffectDataContents)
 			}
 			if err == nil {
-				weh.mutableSideEffect[sideEffectDataID] = &sideEffectDataContents
+				counterHintPayload, ok := attributes.GetDetails()[mutableSideEffectCallCounterName]
+				var counterHint int
+				if ok {
+					err = weh.dataConverter.FromPayloads(counterHintPayload, &counterHint)
```

Comment:
Would suggest eager return here on error (can use different `err` var name for shadowing concerned). Confusing to read code below being executed on error. At the very least add a comment saying you know logic after conditional will run in errored state.